java/arcs/core/entity/Entity.kt

```diff
@@ -15,16 +15,26 @@ import arcs.core.common.Id
 import arcs.core.common.Referencable
 import arcs.core.data.RawEntity
 import arcs.core.data.Schema
+import arcs.core.data.Ttl
 import arcs.core.data.util.ReferencablePrimitive
+import arcs.core.util.Time
 import kotlin.IllegalArgumentException
 import kotlin.reflect.KClass
 
 interface Entity {
     /** The ID for the entity, or null if it is does not have one yet. */
     val entityId: String?
 
-    /** Generates a new ID for the Entity, if it doesn't already have one. */
-    fun ensureIdentified(idGenerator: Id.Generator, handleName: String)
+    /**
+     * Generates a new ID for the Entity, if it doesn't already have one. Also sets creation
+     * timestamp, and expiry timestamp if a ttl is given
+     * */
+    fun ensureIdentified(
```

Comment:
maybe needs a better name now? ensureEntityFields? ensureEntityBase?

Reply:
renamed to ensureEntityFields